packages/syntax/src/Syntax.tsx

```diff
@@ -1,14 +1,56 @@
 import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
-import { css } from '@leafygreen-ui/emotion';
+import flatten from 'lodash/flatten';
+import { cx, css } from '@leafygreen-ui/emotion';
+import { fontFamilies } from '@leafygreen-ui/tokens';
 // Import from core so we can register the appropriate languages ourselves
 import hljs from 'highlight.js/lib/core';
 import hljsDefineGraphQL from 'highlightjs-graphql';
-import CodeWrapper from './CodeWrapper';
-import { Language, SyntaxProps } from './types';
+import { Language, SyntaxProps, Mode } from './types';
 import { SupportedLanguages, languageParsers } from './languages';
 import { injectGlobalStyles } from './globalStyles';
-import renderingPlugin from './renderingPlugin';
+import renderingPlugin, { TableContent } from './renderingPlugin';
+import { SyntaxContext } from './SyntaxContext';
+
+export function expandRangeTuple(tuple: [number, number]): Array<number> {
+  const [lower, upper] = [...tuple]
+    .map(bound => {
+      // Make sure passing infinity doesn't freeze the browser
+      if (bound === Infinity) {
+        // 2000 is arbitrary, but far larger than existing use-cases for our syntax highlighting
+        return 2000;
+      }
+
+      if (bound === -Infinity) {
+        return -2000;
```

Comment:
if this is only used for source code lines, then it'd make sense to clamp the lower-bound to 0, i.e. `bound = Math.max(bound, 0)`

Reply:
I was avoiding bringing in a new dependency to this package, though since it's `lodash`, it probably wouldn't be a huge deal. With how the component is built, there's no added resilience to a clamp here – it does nothing to include a number that doesn't has a corresponding line.

Reply:
no dependency needed; `Math` is a built-in

Reply:
🤦 Duh. Suppose there's no reason not to do this in that case.